runtime/tools/python/ttrt/common/api.py

```diff
@@ -178,13 +178,14 @@ def run(args):
 
     # execution
     print("executing action for all provided flatbuffers")
-    device = ttrt.runtime.open_device(device_ids)
+    # TODO: sort the flatbuffers by runtime type, for now just take the first one
+    ttrt.runtime.set_compatible_runtime(fbb_list[0][1])
+    device = ttrt.runtime.open_device([device_ids[0]])
     atexit.register(lambda: ttrt.runtime.close_device(device))
 
     torch.manual_seed(args.seed)
 
     for (binary_name, fbb, fbb_dict, program_indices) in fbb_list:
-        ttrt.runtime.set_compatible_runtime(fbb)
```

Comment:
@jnie-TT, I'm actually glad we got bit by this right away! What happened was that we called `open_device` above which opens a ttnn device, then we `ttrt.runtime.set_compatible_runtime(fbb)` to ttmetal runtime and boom!

Can you file an issue to track this?  We could do something like:
```c++
enum class DeviceRuntime {    
  Disabled,    
  TTNN,    
  TTMetal,    
}; 

struct ObjectImpl {             
  DeviceRuntime runtime;                                                                                                                                                               
  std::shared_ptr<void> handle;                                                                                                                                                      
                                                                                                                                                                                    
  ObjectImpl(std::shared_ptr<void> handle) : handle(handle) {}                                                                                                                                           
  template <DeviceRuntime R, typename T> T &as() { assert(associatedRuntime == R); return *static_cast<T *>(handle.get()); }                                                                                                                     
  template <DeviceRuntime R, typename T> T const &as() const {                  
    assert(associatedRuntime == R);                                                                                                                      
    return *static_cast<T const *>(handle.get());                                                                                                                                    
  }                                                                                                                                                                             
};
```

Let's just make it impossible to mix types between the two runtime worlds.

Reply:
Ahhhh great catch! I'll file an issue!

Reply:
@nsmithtt great catch! I filed an issue #331